script.js

```diff
@@ -0,0 +1,306 @@
+const suits = [
+    {
+        suit: '♣',
+        color: 'black'
+    },
+    {
+        suit: '♠',
+        color: 'black'
+    },
+    {
+        suit: '♥',
+        color: 'red'
+    },
+    {
+        suit: '♦',
+        color: 'red'
+    }
+]
+
+const faces = [
+    {
+        face: 'A',
+        value: 1
+    },
+    {
+        face: '2',
+        value: 2
+    },
+    {
+        face: '3',
+        value: 3
+    },
+    {
+        face: '4',
+        value: 4
+    },
+    {
+        face: '5',
+        value: 5
+    },
+    {
+        face: '6',
+        value: 6
+    },
+    {
+        face: '7',
+        value: 7
+    },
+    {
+        face: '8',
+        value: 8
+    },
+    {
+        face: '9',
+        value: 9
+    },
+    {
+        face: '10',
+        value: 10
+    },
+    {
+        face: 'J',
+        value: 11
+    },
+    {
+        face: 'Q',
+        value: 12
+    },
+    {
+        face: 'K',
+        value: 13
+    }
+]
+
+const slider = document.getElementById("difficulty")
+const startBtn = document.getElementById("start")
+let side = 'R';
+let difficulty = 0
+let cards = [];
+let drawnCards = [];
+let matches = 0;
+let yourMatches = 0;
+let yourLosses = 0;
+
+function createDeck() {
+    suits.forEach(s => {
+        faces.forEach(f => {
+            let card = {
+                suit: s.suit,
+                color: s.color,
+                face: f.face,
+                value: f.value
+            }
+            cards.push(card);
+        })
+    });
+}
```

Comment:
This function has side-effects for no reason - here you can rewrite it without relying on side effects:

```suggestion
function createDeck(suits, faces) {
    let deck = [];
    suits.forEach(s => {
        faces.forEach(f => {
            let card = {
                suit: s.suit,
                color: s.color,
                face: f.face,
                value: f.value
            }
            deck.push(card);
        })
    });
    return deck;
}
```